bundle/config/mutator/resolve_variable_references.go

```diff
@@ -13,10 +15,50 @@ import (
 
 type resolveVariableReferences struct {
 	prefixes []string
+	pattern  dyn.Pattern
+	lookupFn func(dyn.Value, dyn.Path) (dyn.Value, error)
 }
 
 func ResolveVariableReferences(prefixes ...string) bundle.Mutator {
-	return &resolveVariableReferences{prefixes: prefixes}
+	return &resolveVariableReferences{prefixes: prefixes, lookupFn: lookup}
+}
+
+func ResolveVariableReferencesInLookup() bundle.Mutator {
+	return &resolveVariableReferences{prefixes: []string{
+		"bundle",
+		"workspace",
+		"variables",
```

Comment:
For parity these prefixes should be pushed down into a constructor for the main case as well. Now they are split between the `phases` code and here.

Reply:
We actually use different prefixes in build and deploy phases, so we need to keep prefixes as constructor params